File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack=[]
        for i in range(len(s)):
            if s[i]=="(" or s[i]=="[" or s[i]=="{":
                stack.append(s[i])
            else:
                if len(stack)==0:
                    return False
                last=stack[len(stack)-1]
                stack.pop()
                last+=s[i]
                if(last=="{}" or last=="()" or last=="[]"):
                    continue
                else:
                    return False

        if len(stack)>0:
            return False
        else:
            return True
        

File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    let arr=[]
    let map={
        ")":"(",
        "]":"[",
        "}":"{"
    }

    for(let i=0;i<s.length;i++){
        if(s[i]=="(" || s[i]=="{" ||s[i]=="["){
            arr.push(s[i])
        }
        else{
            let last=arr.pop()
            if(map[s[i]]!=last) return false
        }
    }
    if(arr.length!==0){
        return false
    }
    return true;
};